test/util/util.go

```diff
@@ -167,7 +176,7 @@ func ExpectWorkloadsToBePending(ctx context.Context, k8sClient client.Client, wl
 				continue
 			}
 			cond := updatedWorkload.Status.Conditions[idx]
-			if cond.Status == metav1.ConditionFalse && cond.Reason == "Pending" && wl.Status.Admission == nil {
+			if cond.Status == metav1.ConditionFalse && cond.Reason == "Pending" && !workload.IsWorkloadAdmitted(wl) {
```

Comment:
In theory, this should be enough, because we are checking the WorkloadAdmitted condition.

```suggestion
			if cond.Status == metav1.ConditionFalse && cond.Reason == "Pending" {
```

but this check is fine.